uast/uast.go

```diff
@@ -262,87 +270,311 @@ type Any interface{}
 type Alias struct {
 	GenNode
+	// Name assigned to an entity.
+	//
+	// TODO: define a different node to handle QualifiedIdentifier as a name
 	Name Identifier `json:"Name"`
-	Node Any        `json:"Node"`
+
+	// A UAST node to assign a name to.
+	Node Any `json:"Node"`
+
+	// TODO: should include a pointer to a scope where an alias is defined
 	// Target *Scope
 }
 
+// Import is a statement that can load other modules into the program or library.
+//
+// This is a declarative import statement. It's position in the UAST does not affect
+// the way and the time when the module is imported and the side-effects are executed
+// only once a package is initialized.
+//
+// Similar to import in Go, Java, C#.
+//
+// For more specific types see RuntimeImport, RuntimeReImport, InlineImport.
 type Import struct {
 	GenNode
-	Path   Any   `json:"Path"`
+	// Path is a path of a modules or package to load.
+	//
+	// May have a value of:
+	// - String (specifies relative or absolute module path);
+	// - QualifiedIdentifier (specifies a canonical module name);
+	// - Alias (contains any of the above and defines a local package name within a file/scope);
+	Path Any `json:"Path"`
+
+	// All is set to true when the statement defines all exported symbols from
+	// a module in the local scope (usually file).
 	All    bool  `json:"All"`
 	Names  []Any `json:"Names"`
 	Target Scope `json:"Target"`
 }
 
+// RuntimeImport is a type of an import statement that imports a module only when an execution
+// reaches this UAST node. The import side effects are executed only once, regardless of how many
+// times a statement is reached.
+//
+// Similar to import in PHP, Python, JS.
+//
+// For other import types, see Import.
 type RuntimeImport Import
 
+// RuntimeReImport is a subset of RuntimeImport statement that will re-execute an import and its
+// side-effects statement each time an execution reaches the statement.
+//
+// Similar to import in PHP, Python.
+//
+// For other import types, see Import.
 type RuntimeReImport RuntimeImport
 
+// InlineImport is a subset of import statement that acts like a preprocessor - all statements in
+// the imported module are copied into a position of the UAST node.
+//
+// Similar to import in C and C++.
+//
+// For other import types, see Import.
 type InlineImport Import
 
+// Argument is a named argument or return of a function.
 type Argument struct {
 	GenNode
-	Name        *Identifier `json:"Name"`
-	Type        Any         `json:"Type"`
-	Init        Any         `json:"Init"`
-	Variadic    bool        `json:"Variadic"`
-	MapVariadic bool        `json:"MapVariadic"`
-	Receiver    bool        `json:"Receiver"`
+	// Name is an optional name of an argument.
+	Name *Identifier `json:"Name"`
+
+	// Type is an optional type of an argument.
+	Type Any `json:"Type"`
+
+	// Init is an expression that should be used to initialize an argument
+	// in case no value is provided. Also known as a default value of an argument.
```

Comment:
Subtly, the default value is what the expression yields, whereas `Init` is the expression itself.

Reply:
Right. I will drop the last sentence.